pkg/util/resource/config.go

```diff
@@ -170,6 +172,40 @@ func ParseVolume(item string) (*Config, error) {
 	}, nil
 }
 
+// CreateAndParseVolume will create and parse a volume and return a Config.
+func CreateAndParseVolume(item string) (*Config, error) {
+	pvcParts := strings.Split(item, ":")
+
+	if len(pvcParts) != 4 {
+		return nil, fmt.Errorf("could not create and mount pvc as %s", item)
+	}
+
+	pvc := corev1.PersistentVolumeClaim{
```

Comment:
I think this object must be passed back to the caller. There you must invoke a function to store the PVC just created in the cluster. As an example, see what we do for the Service trait. We create it and then we apply it on the cluster: https://github.com/apache/camel-k/blob/main/pkg/trait/service.go#L112